core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -298,10 +298,10 @@ private[spark] class TaskSchedulerImpl(
       if (!executorIdToHost.contains(o.executorId)) {
```

Comment:
Should we be checking executorIdToTaskCount here? That seems to be what we use everywhere else to detect executor liveness.  It looks like, when a new executor is added, we always add it to both maps (here), and then when it's removed, sometimes we remove it from executorIdToTaskCount before removing it from executorIdToHost (if we're still waiting for the reason the executor was lost). Since we don't reuse executor IDs, AFAIK, it doesn't actually matter which map we check here, so might be good to do executorIdToTaskCount for consistency (and then change line 302 to just update).